fil_fungible_token/src/token/mod.rs

```diff
@@ -76,25 +76,21 @@ where
     ///
     /// Returns a Token handle that can be used to interact with the token state tree and the Cid
     /// of the state tree root
-    pub fn new(bs: BS, msg: MSG) -> Result<(Self, Cid)> {
-        let init_state = TokenState::new(&bs)?;
-        let cid = init_state.save(&bs)?;
-        let token = Self { bs, msg, state: init_state };
-        Ok((token, cid))
-    }
-
-    /// For an already initialised state tree, loads the state tree from the blockstore and returns
-    /// a Token handle to interact with it
-    pub fn load(bs: BS, msg: MSG, state_cid: Cid) -> Result<Self> {
-        let state = TokenState::load(&bs, &state_cid)?;
-        Ok(Self { bs, msg, state })
+    pub fn new(bs: BS, msg: MSG, state: &'st mut TokenState) -> Result<Self> {
+        let token = Self { bs, msg, state };
+        Ok(token)
     }
```

Comment:
If we go this way, I'd expect to add an `init` or similar method that creates new state as the old `new` did, and returns both. And similarly for `load` probably.

Reply:
I think this is not possible anymore the way it's currently structured (@abright maybe you can see a way perhaps with heap allocation via a `Box`). The awkwardness of _always_ having to create the `TokenState` externally and pass it in to a `Token` is the primary reason I turned to a `RefCell` impl

If a helper function instantiates the `TokenState` that token state is dropped at the end of whatever helper function creates it. Therefore the lifetime of any reference to it becomes invalid and can't be used in the `Token` that we actually want returned.

Reply:
But it won't be dropped if it's returned....

Reply:
I guess you won't be able to return the Token then. Ok fine, consider it just a convenience for creating a state.

Reply:
I'm not sure if this is what you are thinking of https://github.com/helix-onchain/filecoin/pull/60/commits/3b330574f72d31fa5991f7c02fd6ec2065411b54